cpp/src/plasma/CMakeLists.txt

```diff
@@ -52,6 +52,29 @@ if(FLATBUFFERS_VENDORED)
   add_dependencies(gen_plasma_fbs flatbuffers_ep)
 endif()
 
+externalproject_add(asio_ep
```

Comment:
This should be formatted more like so to be consistent with `ThirdpartyToolchain.cmake`. Maybe even consider to add it there?
```
ExternalProject_Add(asio_ep
  URL https://github.com/chriskohlhoff/asio/archive/asio-1-12-2.zip
  CONFIGURE_COMMAND "" # No autogen since we use asio in header-only way.
  BUILD_IN_SOURCE 1
  BUILD_COMMAND ""
  INSTALL_COMMAND cmake -E copy asio/include/asio.hpp ${OUTPUT_DIR}/.. &&
                                      cmake -E copy_directory asio/include/asio ${OUTPUT_DIR}/../asio/)
```

Reply:
It was formatted by the cmake format checker. I can move it into `ThirdpartyToolchain.cmake` to keep the currect format.

Reply:
Can you hang on until the CMake refactor is merged to make sure this doesn't conflict (easier to fix this PR than to fix that)?

Reply:
Sure. Which PR is about CMake refactor?

Reply:
https://github.com/apache/arrow/pull/3688